### PR TITLE
Fixed home.dart (issue #68)

### LINKED ISSUE #68: Overflow of widgets on Home screen
# Description
Some widgets on the home screen overflow by some pixels. This occurs on 1080p and 720p screen resolution devices and does not occur on 1440p devices (tested on only these resolutions).

# Relevant files
lib/Screens/Home.dart

# Screenshot
Screenshot_1638961295

### PR DESCRIPTION
**Pull request by raviyadav17**
## Issue
Fixes #68 

## Description

I have updated a little part of the code and fixed the error.

## Type of change

-  Bug fix (non-breaking change which fixes an issue)

## Screenshot
Screenshot_1639140528

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] The app runs properly
- [ ] I have performed a self-review of my own code
- [ ] I have formatted and commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

I am member of Codepeak 
thank you

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
@raviyadav17, please merge master of my repository

**Comment by raviyadav17** [comment]
I have merged the changes to the master branch of my repository . I don't know how to merge to your master branch.

**Comment by Saransh-cpp** [comment]
@raviyadav17, you don't have to merge into my master branch. Merge my master branch into your PR's branch. If you are having any issues then please let me know the output of `git remote -v`.

**Comment by raviyadav17** [comment]
Can you tell how to Merge my master branch into your PR's branch??

**Comment by Saransh-cpp** [comment]
What does `git remote -v` give you?

**Comment by raviyadav17** [comment]
origin  https://github.com/raviyadav17/BookRentApp-Chapter3.git (fetch)
origin  https://github.com/raviyadav17/BookRentApp-Chapter3.git (push)

**Comment by Saransh-cpp** [comment]
Make sure you are on your `Fixed-Home.dart` branch, execute the following -
```
git remote add upstream https://github.com/Saransh-cpp/BookRentApp-Chapter3
git pull upstream master
```
This will add a new remote named upstream which will point to my repository (you can check using `git remote -v`). The second command will merge my master branch to your current branch.

After this, resolve the conflicts and push the code.

**Comment by raviyadav17** [comment]
**Done**
origin  https://github.com/raviyadav17/BookRentApp-Chapter3.git (fetch)
origin  https://github.com/raviyadav17/BookRentApp-Chapter3.git (push)
upstream        https://github.com/Saransh-cpp/BookRentApp-Chapter3 (fetch)
upstream        https://github.com/Saransh-cpp/BookRentApp-Chapter3 (push)

**But** 
CONFLICT (content): Merge conflict in lib/Screens/Register.dart
Automatic merge failed; fix conflicts and then commit the result.

**Comment by Saransh-cpp** [comment]
Yes, you have to fix these conflicts in your code manually, in the file `lib/Screens/Register.dart`.

**Comment by raviyadav17** [comment]
> Yes, you have to fix these conflicts in your code manually, in the file `lib/Screens/Register.dart`.

I am trying but lint-check is failing everytime

File: lib/Screens/Register.dart [hunk 1]
```diff
@@ -1,85 +1,101 @@
-//import 'dart:math';
 
-//import 'package:firebase_database/firebase_database.dart';
+import 'dart:math';
+
+import 'package:firebase_database/firebase_database.dart';
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:4]:
```suggestion
// import 'package:firebase_database/firebase_database.dart';
```

File: lib/Screens/Register.dart [hunk 2]
```diff
@@ -1,85 +1,101 @@
-//import 'dart:math';
 
-//import 'package:firebase_database/firebase_database.dart';
+import 'dart:math';
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:2]:
```suggestion
// import 'dart:math';
```

File: lib/Screens/Register.dart [hunk 3]
```diff
@@ -1,85 +1,101 @@
...
 import 'package:provider/provider.dart';
 import 'package:test_app/Screens/Loading.dart';
 //import 'package:fluttertoast/fluttertoast.dart';
 import 'package:test_app/Screens/NavBar.dart';
-//import 'package:test_app/Utility/auth.dart';
-//import 'package:test_app/main.dart';
-//import 'dart:io';
+import 'package:test_app/Utility/auth.dart';
+import 'package:test_app/main.dart';
+import 'dart:io';
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:11]:
```suggestion
// import 'package:test_app/Utility/auth.dart';
// import 'package:test_app/main.dart';
// import 'dart:io';
```

File: lib/Screens/Register.dart [hunk 4]
```diff
@@ -1,85 +1,101 @@
...
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
-//import 'package:image_picker/image_picker.dart';
-//import 'package:shared_preferences/shared_preferences.dart';
-//import 'package:test_app/Screens/LoginPage.dart';
-//import 'package:firebase_storage/firebase_storage.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:test_app/Screens/LoginPage.dart';
+import 'package:firebase_storage/firebase_storage.dart';
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:18]:
```suggestion
// import 'package:image_picker/image_picker.dart';
// import 'package:shared_preferences/shared_preferences.dart';
// import 'package:test_app/Screens/LoginPage.dart';
// import 'package:firebase_storage/firebase_storage.dart';
```

File: lib/Screens/Register.dart [hunk 5]
```diff
@@ -1,85 +1,101 @@
...
   _RegisterState createState() => _RegisterState();
 }
 
 class _RegisterState extends State<Register> {
+
   //Santos
   final FirebaseAuth firebaseAuth = FirebaseAuth.instance;
   final _formKey = GlobalKey<FormState>();
-  final _key = GlobalKey<ScaffoldMessengerState>();
+  final _key = GlobalKey<ScaffoldState>();
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:36]:
```suggestion
  final _key = GlobalKey<ScaffoldMessengerState>();
```

File: lib/Screens/Register.dart [hunk 6]
```diff
@@ -1,85 +1,101 @@
...
   TextEditingController _emailTextController = TextEditingController();
   TextEditingController _passwordTextController = TextEditingController();
   TextEditingController _nameTextController = TextEditingController();
   TextEditingController _confirmPasswordController = TextEditingController();
 
+
   String email = '';
   String password = '';
-  String cpassword = '';
+  String cpassword ='';
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:46]:
```suggestion
  String cpassword = '';
```

File: lib/Screens/Register.dart [hunk 7]
```diff
@@ -1,85 +1,101 @@
...
   String error = '';
   String name = '';
   bool hidePass1 = true;
   bool hidePass2 = true;
   String userImageUrl = '';
+  File _imageFile;
+
+
+  @override
+
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:56]:
```suggestion
```

File: lib/Screens/Register.dart [hunk 8]
```diff
@@ -1,85 +1,101 @@
...
   String email = '';
   String password = '';
-  String cpassword = '';
+  String cpassword ='';
   String error = '';
   String name = '';
   bool hidePass1 = true;
   bool hidePass2 = true;
   String userImageUrl = '';
+  File _imageFile;
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:52]:
```suggestion
```

File: lib/Screens/Register.dart [hunk 9]
```diff
@@ -1,85 +1,101 @@
...
+  @override
+
 
-  Widget emailAndNameField(String hint, String error, Icon icon) {
-    double _screenWidth = MediaQuery.of(context).size.width;
+  Widget EmailAndNameField(String hint , String error , Icon icon){
+    double _screenWidth = MediaQuery
+        .of(context)
+        .size
+        .width;
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:62]:
```suggestion
  Widget emailAndNameField(String hint, String error, Icon icon) {
    double _screenWidth = MediaQuery.of(context).size.width;
```

File: lib/Screens/Register.dart [hunk 10]
```diff
@@ -1,85 +1,101 @@
...
     return Column(
       children: [
         Container(
-          width: _screenWidth / 1.08,
+          width: _screenWidth/1.08,
```
Comment (Saransh-cpp) [inline lib/Screens/Register.dart:66]:
```suggestion
          width: _screenWidth / 1.08,
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
@raviyadav17, I have listed some of the changes below

There are changes in files other than `Home.dart` (there shouldn't be any as this PR is only concerned with `Home.dart`). Hence you should revert all of them back, manually.

The linting issue should be solved once all the other files are reverted to their initial state.

Let me know if you face any issues! :)

**Comment by raviyadav17** [comment]
Still lint check is failed after reverting changes

**Comment by Saransh-cpp** [comment]
@raviyadav17, I pointed out only a few issues. You have to revert back all the files to their initial state, except `Home.dart`.

I see 3 files changes in this PR, but this PR should only change the `Home.dart` file -

image

**Pull request closed without merging**